File: conformance/tests/post-quantum-wots/post-quantum-wots.runar.zig
const runar = @import("runar");

pub const PostQuantumWOTS = struct {
    pub const Contract = runar.SmartContract;

    pubkey: runar.ByteString,

    pub fn init(pubkey: runar.ByteString) PostQuantumWOTS {
        return .{ .pubkey = pubkey };
    }

    pub fn spend(self: *const PostQuantumWOTS, msg: runar.ByteString, sig: runar.ByteString) void {
        runar.assert(runar.verifyWOTS(msg, sig, self.pubkey));
    }
};

File: conformance/tests/post-quantum-wots/post-quantum-wots.runar.go
//go:build ignore

package contract

import "runar"

type PostQuantumWOTS struct {
	runar.SmartContract
	Pubkey runar.ByteString `runar:"readonly"`
}

func (c *PostQuantumWOTS) Spend(msg runar.ByteString, sig runar.ByteString) {
	runar.Assert(runar.VerifyWOTS(msg, sig, c.Pubkey))
}
